Mathlib/Order/SuccPred/LinearLocallyFinite.lean

```diff
@@ -165,6 +171,18 @@ instance (priority := 100) LinearOrder.isPredArchimedean_of_isSuccArchimedean [S
       · rw [hn_eq]
         exact hn_lt_ne _ (Nat.lt_succ_self n)
 
+-- We don't make this an instance to avoid loops with `isPredArchimedean_of_isSuccArchimedean`.
+theorem isSuccArchimedean_of_isPredArchimedean [IsPredArchimedean ι] : IsSuccArchimedean ι :=
+  inferInstanceAs (IsSuccArchimedean ιᵒᵈᵒᵈ)
```

Comment:
I find this design really weird. Why would one be an instance but not the other? The expectation is that either they are both instances or they are both non-instances. Also note that instance loops are *supposed* to be fine in Lean 4

Reply:
I assumed that avoiding the instance loop was the reason this instance hadn't been declared in the first place, though maybe that was just a holdover from Lean 3. I'll turn it into an instance then.